pkg/apis/eventing/v1alpha1/trigger_types.go

```diff
@@ -58,9 +58,19 @@ type TriggerSpec struct {
 	// +optional
 	DeprecatedGeneration int64 `json:"generation,omitempty"`
 
-	Broker     string          `json:"broker,omitempty"`
+	Broker string `json:"broker,omitempty"`
+
+	// +optional
+	Filter *TriggerFilter `json:"filter,omitempty"`
+
 	Subscriber *SubscriberSpec `json:"subscriber,omitempty"`
+}
 
+type TriggerFilter struct {
+	ExactMatch *TriggerFilterAttributes `json:"exactMatch,omitempty"`
```

Comment:
On the theory that we will eventually support exact matching other attributes, how about naming this `TypeAndSource`?